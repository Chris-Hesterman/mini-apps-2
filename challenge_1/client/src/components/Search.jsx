import React from 'react';
import styled from 'styled-components';

const StyledInput = styled.input`
  background: rgba(230, 230, 230, 0.5);
  border-color: #d1913c;
  font-size: 18px;
  height: 25px;
  width: 175px;
  outline: none;
  margin-bottom: 40px;
  border-radius: 5px;
`;

const StyledButton = styled.button`
  box-sizing: border-box;
  padding: 4px;
  margin-left: 25px;
  font-size: 1rem;
  font-weight: bold;
  border: 2px solid #d1913c;
  border-radius: 5px;
  box-shadow: 2px 2px 2px grey;
  outline: none;
  &:active {
    transform: translate(1px, 1px);
    box-shadow: 1px 1px 1px grey;
  }
`;
class Search extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      inputValue: ''
    };
    this.onChange = this.onChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  onChange(e) {
    console.log(this.state.inputValue);
    this.setState({ inputValue: e.target.value });
  }

  handleSubmit(e) {
    e.preventDefault();
    this.props.send(this.state.inputValue);
    this.setState({ inputValue: '' });
  }

  render() {
    return (
      <div>
        <form onSubmit={this.handleSubmit}>
          <StyledInput
            value={this.state.inputValue}
            type="text"
            onChange={this.onChange}
            placeholder="Enter search term"
            autoFocus
          />
          <StyledButton type="submit">Submit</StyledButton>
        </form>
      </div>
    );
  }
}

export default Search;
