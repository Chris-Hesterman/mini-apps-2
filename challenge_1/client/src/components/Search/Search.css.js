import styled from 'styled-components';

// Search.jsx styles ------
export const StyledInput = styled.input`
  background: rgba(230, 230, 230, 0.5);
  border-color: #d1913c;
  font-size: 18px;
  height: 25px;
  width: 175px;
  outline: none;
  border-radius: 5px;
`;

export const StyledButton = styled.button`
  box-sizing: border-box;
  padding: 6px 5px 4px 5px;
  margin-left: 25px;
  font-size: 1rem;
  font-weight: bold;
  border: 2px solid #d1913c;
  border-radius: 5px;
  background: rgba(230, 230, 230, 0.5);
  box-shadow: 2px 2px 2px grey;
  outline: none;
  &:active {
    transform: translate(1px, 1px);
    box-shadow: 1px 1px 1px grey;
  }
`;
