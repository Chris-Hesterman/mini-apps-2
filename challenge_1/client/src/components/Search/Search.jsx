import React from 'react';
import styled from 'styled-components';
import { StyledInput, StyledButton } from './Search.css.js';

class Search extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      inputValue: ''
    };
    this.onChange = this.onChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  onChange(e) {
    this.setState({ inputValue: e.target.value });
  }

  handleSubmit(e) {
    e.preventDefault();
    this.props.send(this.state.inputValue, 1);
    this.setState({ inputValue: '' });
  }

  render() {
    return (
      <div>
        <form onSubmit={this.handleSubmit}>
          <StyledInput
            value={this.state.inputValue}
            type="text"
            onChange={this.onChange}
            placeholder="Enter search term"
            autoFocus
          />
          <StyledButton type="submit">Submit</StyledButton>
        </form>
      </div>
    );
  }
}

export default Search;
