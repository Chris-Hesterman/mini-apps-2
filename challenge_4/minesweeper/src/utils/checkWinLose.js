export const checkWin = (uncovered, mines) => {
  if (uncovered.size === 100 - mines.length) {
    return true;
  }

  return false;
};

export const checkLose = (squareNum, mines, flags) => {
  if (mines.includes(squareNum) && !flags.includes(squareNum)) {
    return true;
  }
  return false;
};
