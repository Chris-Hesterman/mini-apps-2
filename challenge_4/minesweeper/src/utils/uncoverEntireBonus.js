import checkAdjacent from './checkAdjacent';

export const uncoverEntireBonus = (number, uncoveredSquares, minesArray) => {
  let fullBonus = [];

  const helper = (num) => {
    let arr = buildAdjacent(num, uncoveredSquares, fullBonus);

    fullBonus = fullBonus.concat(arr);
    arr = arr.forEach((number) => {
      if (checkAdjacent(number, minesArray) === 0) {
        helper(number);
      }
    });
  };
  helper(number);

  return new Set(fullBonus);
};

export const buildAdjacent = (num, uncovered, bonus) => {
  let adjArray = [num - 10, num + 10];

  if (num % 10 !== 0 && num % 10 !== 1) {
    adjArray.push(num - 1, num + 1, num - 9, num + 9, num - 11, num + 11);
  }

  if (num % 10 === 1) {
    adjArray.push(num + 1, num - 9, num + 11);
  }

  if (num % 10 === 0) {
    adjArray.push(num + 9, num - 1, num - 11);
  }

  adjArray = adjArray.filter((number) => {
    if (uncovered.length) {
      if (
        number > 0 &&
        number < 101 &&
        uncovered.indexOf(number) === -1 &&
        !bonus.includes(number)
      ) {
        return number;
      }
    } else {
      if (number > 0 && number < 101) {
        return number;
      }
    }
  });

  return adjArray;
};
