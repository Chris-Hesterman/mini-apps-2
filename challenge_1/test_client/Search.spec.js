import React from 'react';
import PropTypes from 'prop-types';
import { checkProps } from '../testUtils/utils.js';
import Enzyme from 'enzyme';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import Search from '../client/src/components/Search/Search.jsx';
import { findByTestAttr } from '../testUtils/utils.js';

Enzyme.configure({ adapter: new Adapter() });

describe('Search component', () => {
  describe('Checking propTypes', () => {
    it('Should not throw a warning', () => {
      const expectedProps = {
        send: () => 'send query'
      };
      const propsError = checkProps(Search, expectedProps);

      expect(propsError).toBe(undefined);
    });
  });
  describe('snapshot testing', () => {
    it('renders correctly', () => {
      const tree = renderer.create(<Search />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
