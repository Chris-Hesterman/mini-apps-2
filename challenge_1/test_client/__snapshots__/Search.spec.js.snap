// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component snapshot testing renders correctly 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      className="Searchcss__StyledInput-sc-21w6vx-0 gzayPf"
      onChange={[Function]}
      placeholder="Enter search term"
      type="text"
      value=""
    />
    <button
      className="Searchcss__StyledButton-sc-21w6vx-1 lbZijS"
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
